### PR TITLE
Show indicator for currently selected subclass path

### LINKED ISSUE #4201: Show subclass on classes
This would require some mapping, but it'd be great to show which subclass is active on the class icon. 

@inexorableAce do you know what icons we could use? I guess we could use text as well.

### PR DESCRIPTION
**Pull request by vivekhnz**
This PR is an evolution of #4264 and closes #4201.

dim-subclass-path-indicators
dim-subclass-path-indicators-2

The main changes are:

- Replace the subclass icon with the super icon for the actively selected subclass path
- Display a small diamond in a corner of the subclass icon to indicate which subclass path is actively selected

Still to do:

- [x] Replace hardcoded URIs with talent grid icons from API

### CODE REVIEW

**Comment by bhollis** [comment]
This looks so good!!

**Comment by inexorableAce** [comment]
This is dope! I'll try to have the vectors done this weekend!

**Comment by vivekhnz** [comment]
@inexorableAce and I did some tests with vectorised versions of the super icons and found that the PNGs from the API often looked better than the SVGs when scaled down. The current plan is to get a version of this into beta with PNGs and get some feedback across a wider range of screens. We can revisit SVGs if they seem necessary.

File: src/app/inventory/ConnectedInventoryItem.tsx [hunk 1]
```diff
@@ -30,7 +31,7 @@ interface StoreProps {
   inventoryCuratedRoll?: InventoryCuratedRoll;
 }
 
-function mapStateToProps(state: RootState, props: ProvidedProps): StoreProps {
+function mapStateToProps(state: RootState, props: ProvidedProps): Props {
```
Comment (bhollis) [inline src/app/inventory/ConnectedInventoryItem.tsx]: This should always return only `StoreProps` - that interface is meant to represent the subset of props which are returned from `mapStateToProps`.

File: src/app/inventory/ConnectedInventoryItem.tsx [hunk 2]
```diff
@@ -39,6 +40,7 @@ function mapStateToProps(state: RootState, props: ProvidedProps): StoreProps {
   const showRating = shouldShowRating(dtrRating);
 
   return {
+    ...props,
```
Comment (bhollis) [inline src/app/inventory/ConnectedInventoryItem.tsx]: There's no need to mix props into this - they are already added in by `connect`.

File: src/app/inventory/ConnectedInventoryItem.tsx [hunk 3]
```diff
@@ -82,9 +85,12 @@ function ConnectedInventoryItem({
...
       curationEnabled={curationEnabled}
       inventoryCuratedRoll={inventoryCuratedRoll}
+      doNotRepresentSelectedPerks={doNotRepresentSelectedPerks}
       innerRef={innerRef}
     />
   );
 }
 
-export default connect<StoreProps>(mapStateToProps)(ConnectedInventoryItem);
+export default connect<Props, {}, ProvidedProps, RootState>(mapStateToProps)(
```
Comment (bhollis) [inline src/app/inventory/ConnectedInventoryItem.tsx]: This signature doesn't need to change.

File: src/app/inventory/InventoryItem.tsx [hunk 4]
```diff
@@ -32,6 +35,8 @@ interface Props {
   searchHidden?: boolean;
   curationEnabled?: boolean;
   inventoryCuratedRoll?: InventoryCuratedRoll;
+  /** Don't show information that relates to currently selected perks */
```
Comment (bhollis) [inline src/app/inventory/InventoryItem.tsx]: Mention that this is just for subclasses?

File: src/app/inventory/InventoryItem.tsx [hunk 5]
```diff
@@ -72,28 +73,41 @@ export default function InventoryItem({
     };
   }
 
+  const subclassPath =
+    (!doNotRepresentSelectedPerks &&
```
Comment (bhollis) [inline src/app/inventory/InventoryItem.tsx]: I generally don't like double negatives

File: src/app/item-popup/ItemTalentGrid.tsx [hunk 6]
```diff
@@ -106,6 +106,7 @@ function ItemTalentGrid({ item, perksOnly, bestPerks }: Props) {
                 r="16"
                 cx="-17"
                 cy="17"
+                id={node.hash.toString()}
```
Comment (bhollis) [inline src/app/item-popup/ItemTalentGrid.tsx]:
```suggestion
```

Comment (bhollis) [reply src/app/item-popup/ItemTalentGrid.tsx]: Sorry, testing out the GitHub suggested changes thing. I'd added this to make it easier to figure out the perks for subclasses, but we don't need it anymore.

Comment (vivekhnz) [reply src/app/item-popup/ItemTalentGrid.tsx]: Thanks, I've removed it 👍

**Pull request merged**